src/main/java/ReviewInteractors/DeleteReviewInteractor.java

```diff
@@ -0,0 +1,54 @@
+//Class responsible for deleting reviews from the user's and restaurant's list of reviews + databases
+
+package ReviewInteractors;
+
+import ReviewInterfaces.DeleteReviewInputBoundary;
+import ReviewScreens.ReviewResponseModel;
+import entities.Restaurant;
+import entities.Review;
+import entities.User;
+import restaurant_use_case.RestaurantDSRequestModel;
+
+import java.io.IOException;
+
+public class DeleteReviewInteractor implements DeleteReviewInputBoundary {
+
+    /*
+    Method to delete review from the necessary databases
+     */
+    public ReviewResponseModel interact(DeleteReviewRequestModel requestModel){
+        try {
+            //Get objects and store for ease of reading
+            Review review = requestModel.getReview();
+            String reviewID = review.getID();
+            User user = requestModel.getUser();
+            Restaurant restaurant = requestModel.getRestaurant();
+
+            //Set the Review's visible to false and reflect that change in the database
+            review.setVisible(false);
+            requestModel.getReviewGateway().deleteReview(reviewID);
```

Comment:
If the review is set to invisible, shouldn't the database not delete, but rather just set invisible as well?

Right now if I'm not mistaken just after setting this review to be invisible and then deleting it, it would not exist unless you continue to pass it through (cannot reinitialize from database)